snowballDevs/kandoo
# Added functionality so User can now select a board and navigate to workspace

## Description

#### Quite a big PR so get ready.
- Created axios methods within data service to allow crud operations on the frontend
- Updated the Create a board form to suit.
- Added Kandoo logo to public folder
- Header component now lives within each page/feature (landingpage, dashboard, workspace) instead of the root
- Added a tailwind plugin for forms - https://github.com/tailwindlabs/tailwindcss-forms
- User can now create a board, which will then take them straight to the workspace 
- User can now delete a board, by clicking 'X' within board card
- Created a boardContext, to hold the selectedBoard choosen by the user
- Created Workspace feature
- Created Page Header component, which can hold links/ show information based upon what page user is on


## Related Issue

-   [x] Is this related to a specific issue? Is so please specify: #75 #73 

## Type of Changes

Use a check for ✓ the following type of changes:

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓  | :sparkles: New feature     |
|   ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### After

### Updated Dashboard, users can now click on selected board and go into workspace
![image](https://github.com/snowballDevs/kandoo/assets/105207353/3665b411-3b44-4531-9adf-b24e61d113f7)

### Workspace 
![image](https://github.com/snowballDevs/kandoo/assets/105207353/438973fb-da7c-4543-8f0b-72207e911b5f)

## Testing Steps / QA Criteria

I tested the curd operations in postman first and when ui setup on the interface.

With the routingContext, and selectedBoardContext i used react dev tools to make sure state was being updated correctly

## Checklist:
